src/Utils.js

```diff
@@ -12,6 +12,10 @@ import {
   REACT15,
 } from './version';
 
+export const ITERATOR_SYMBOL = (
+  typeof Symbol === 'function' && Symbol.iterator
+) || '@@iterator';
```

Comment:
We should absolutely not fall back to a string "@@iterator" unless it's the specific versions of Firefox where this imbues magic behavior - which can only be determined by using `for..of` inside a `Function` (so that the syntax doesn't break older engines).